### PR TITLE
[VBLOCKS-2552] Upgrade test app

### PR DESCRIPTION
**Pull request by mhuynh5757**
## Submission Checklist

 - [x] Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] Tested code changes and observed expected behavior in the example app
 - [x] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Description

This PR bootstraps a new React Native application into the `test/` folder. This new test app is bootstrapped with a modern supported version of React Native, and intends to replace the existing test app located at `test/app/`. In future PRs, the app at `test/app/` will be entirely removed and replaced with the new one. As of this PR, they both exist in parallel.

## Breakdown

- Bootstrap new RN application.
- Remove files from the bootstrapping that are not applicable to this project, such as the `tsconfig.json` since we already have one.
- Layer the native changes we've made to the existing test app onto the new test app. This includes things like build configurations and native Java code.
- Copy new dependencies from the new test app to the root project as dev dependencies.
- Satisfy new peer dependencies.

## Validation

- Ran new test app on both Android emulator and a real iOS device.

## Additional Notes

N/A

### CODE REVIEW

File: test/TwilioVoiceReactNativeExample/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+```bash
+# using npm
+npm run android
+
+# OR using Yarn
+yarn android
+```
+
+### For iOS
```
Comment (bobiechen-twilio) [inline test/TwilioVoiceReactNativeExample/README.md:35]: Might need a step to run `pod install`

File: test/TwilioVoiceReactNativeExample/ios/TwilioVoiceReactNativeExample.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -0,0 +1,719 @@
...
+					"$(inherited)",
+					"@executable_path/Frameworks",
+				);
+				MARKETING_VERSION = 1.0;
+				OTHER_LDFLAGS = (
+					"$(inherited)",
+					"-ObjC",
+					"-lc++",
+				);
+				PRODUCT_BUNDLE_IDENTIFIER = com.twilio.TwilioVoiceReactNativeExample;
```
Comment (bobiechen-twilio) [inline test/TwilioVoiceReactNativeExample/ios/TwilioVoiceReactNativeExample.xcodeproj/project.pbxproj:505]: Would be nice to remove the explicit bundle ID.
Not blocking this PR and can be done later (might require some programmatic replace during the build process)

**Review by bobiechen-twilio — APPROVED** [review]
iOS part looks good to me in general

**Review by charliesantos — APPROVED** [review]
Should we delete the test/app folder?

**Comment by mhuynh5757** [comment]
> Should we delete the test/app folder?

I will do this in a future PR as to limit the scope of this one.

**Review by afalls-twilio — APPROVED** [review]

**Comment by charliesantos** [comment]
> > Should we delete the test/app folder?
> 
> I will do this in a future PR as to limit the scope of this one.

Cool. Which ticket is this captured on?

**Comment by mhuynh5757** [comment]
Just made this ticket to track it: https://issues.corp.twilio.com/browse/VBLOCKS-2608

**Pull request merged**